### PR TITLE
Update Xamarin sample code

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Update InferenceSample Xamarin components so that mobile app is more user friendly. Previously it just wrote to Console and had a confusing static UI that didn't make it clear what was happening. Now it has a button to run the tests, and progress updates. 

Remove fasterrcnn xamarin sample. Adding a more complete xamarin sample targeted at end users to the inference examples repo.
  - See https://github.com/microsoft/onnxruntime-inference-examples/pull/58

**Motivation and Context**
Cleanup the Xamarin sample setup.

### CODE REVIEW

File: csharp/sample/InferenceSample/Microsoft.ML.OnnxRuntime.InferenceSample.Forms/MainPage.xaml.cs [hunk 1]
```diff
@@ -18,34 +21,79 @@ protected override void OnAppearing()
...
+        {
+            Action<Label, string> addOutput = (label, text) =>
+            {
+                Device.BeginInvokeOnMainThread(() => { label.Text += text; });
+                Console.Write(text);
+            };
+
+            OutputLabel.Text = "Testing execution\nComplete output is written to Console in this trivial example.\n\n";
+
+            await Task.Run(() =>
```
Comment (edgchen1) [inline csharp/sample/InferenceSample/Microsoft.ML.OnnxRuntime.InferenceSample.Forms/MainPage.xaml.cs:39]: why do we need to put the logic in multiple `Task.Run()`s?

Comment (skottmckay) [reply csharp/sample/InferenceSample/Microsoft.ML.OnnxRuntime.InferenceSample.Forms/MainPage.xaml.cs:40]: Good point - a single task is fine. Updated.

File: csharp/sample/InferenceSample/readme.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Only the native build for the platform you are testing on is required. 
+e.g. if you're testing using an Android device that is arm64, you only need the libonnxruntime.so for arm64-v8a.
+The version of the native build should match the checked-out version of the ONNX Runtime repository you're currently using as closely as possible. 
+Otherwise mismatches with the native entry points is possible and could cause crashes. 
+
+To acquire the native build you can:
+  - build it yourself
+    - [Android](https://onnxruntime.ai/docs/build/android.html) build instructions
+    - [iOS](https://onnxruntime.ai/docs/build/ios.html) build instructions
+  - extract if from the Microsoft.ML.OnnxRuntime nuget package using [NuGetPackageExplorer](https://github.com/NuGetPackageExplorer/NuGetPackageExplorer)
```
Comment (edgchen1) [inline csharp/sample/InferenceSample/readme.md:12]:
```suggestion
  - extract it from the Microsoft.ML.OnnxRuntime nuget package using [NuGetPackageExplorer](https://github.com/NuGetPackageExplorer/NuGetPackageExplorer)
```

File: csharp/sample/InferenceSample/readme.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - release version is [here](https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime/) 
+    - integration test version is [here](https://int.nugettest.org/packages/Microsoft.ML.OnnxRuntime/)
+      - this is frequently update and should work if you're currently using the `master` branch of ONNX Runtime
```
Comment (edgchen1) [inline csharp/sample/InferenceSample/readme.md:15]:
```suggestion
      - this is frequently updated and should work if you're currently using the `master` branch of ONNX Runtime
```

**Review by edgchen1 — DISMISSED** [review]

**Review by gwang-msft — APPROVED** [review]

**Pull request merged**